### PR TITLE
GH-164: [refactor] Enhance Preferred Sports Management with Sport Levels in User Profiles

### PR DESCRIPTION
**Pull request by walidoow**
### Description

This PR introduces enhancements to the management of preferred sports in user profiles by incorporating a new `SportLevel` structure. The changes allow each sport to have an associated name and ranking, providing a more detailed representation of users' sports preferences. Additionally, the PR includes refactoring to improve maintainability and standardize data transformations using mappers.

### Changes

1. **Controller Updates**
- Refactored return types in `AuthController` and `UserController` to directly utilize mappers for improved separation of concerns.
- Adjusted method summaries and descriptions for better clarity in Swagger documentation.

2. **DTO Additions and Updates**
- Introduced `SportLevelRequest` and `SportLevelResponse` DTOs to represent sports preferences with levels.
- Updated `ProfileRequest` and `ProfileResponse` to use `List<SportLevel>` instead of `List<String>` for `sportsOfPreference`.

3. **Mapper Integration**
- Added `ProfileMapper` and `UserMapper` to simplify the conversion between entity and DTO layers.
- Implemented mapping strategies for `updateProfileFromRequest` and `patchProfileFromRequest` to handle partial updates.

4. **Service Layer Refactoring**
- Updated `AuthService` and `UserService` methods to directly return DTO responses using mappers.
- Enhanced profile update and patch logic with a focus on reusability and consistent DTO handling.

5. **Model Enhancements**
- Added a `SportLevel` model to represent individual sports preferences with ranking.
- Updated the `Profile` model to replace `List<String>` with `List<SportLevel>` for sports preferences.

6. **Testing Improvements**
- Refactored `UserResponseTest` to use `UserMapper` for creating DTOs.
- Simplified assertions and introduced additional test cases to cover edge cases.
- Removed redundant `MongoDBContainer` setup in `UserServiceApplicationTests` in favor of simplified context loading.

7. **Miscellaneous Updates**
- Adjusted `application.properties` to include a fallback URI for MongoDB configuration.

### CODE REVIEW

File: Microservices/user-service/src/main/resources/application.properties
Comment (DanDuguay) [inline Microservices/user-service/src/main/resources/application.properties:1]: Did you mean to commit the mongoDB URI?

**Review by DanDuguay — COMMENTED** [review]
Looks good, just want to check if the MongoDB URI is supposed to be kept secret.

**Review by patrickmac3 — APPROVED** [review]
Looks good to me !

File: Microservices/user-service/src/main/resources/application.properties
Comment (walidoow) [reply Microservices/user-service/src/main/resources/application.properties:1]: It is just a dummy URI, a fallback value for unit test to run properly. It doesn't point to anything real

Comment (DanDuguay) [reply Microservices/user-service/src/main/resources/application.properties:1]: understood, Thanks!

**Review by DanDuguay — APPROVED** [review]

**Pull request merged**